### ISSUE TITLE
Add support for nested JWE tokens

### ISSUE THREAD

**Issue by yuriyz** [GluuFederation/oxAuth#949]
Add support for nested JWE tokens
Here is example: https://connect2id.com/products/nimbus-jose-jwt/examples/signed-and-encrypted-jwt

https://support.gluu.org/authentication/6108/problem-with-decrypting-jwe/#at41701

```
The code I added:
//create signed idToken to nest it in jwe payload
jwe.setSignedJWTPayload(generateSignedIdToken(authorizationGrant, nonce, authorizationCode, accessToken, scopes, includeIdTokenClaims, preProcessing));
So I create signed Id Token using the existing signing method and I set it in signedJWTPayload -field.
Please inform us if you can do the change in some future version.
```

**Comment by yuriyz** [comment]
Diff:
```
if (jwr instanceof Jwe) {
        if (client.getIdTokenSignedResponseAlg() != null) {
            JwtSigner jwtSigner = JwtSigner.newJwtSigner(appConfiguration, webKeysConfiguration, client);
            Jwt jwt = jwtSigner.newJwt();
            jwt.setClaims(jwr.getClaims());
            ((Jwe) jwr).setSignedJWTPayload(signJwt(jwt, client));
        }
        return encryptJwe((Jwe) jwr, client);
   }
```

**Comment by yuriyz** [comment]
Implemented in 4.2.2 and jans.
